drv/stm32xx-i2c/src/lib.rs

```diff
@@ -148,33 +159,48 @@ pub enum ReadLength {
     Variable,
 }
 
+#[allow(clippy::upper_case_acronyms)]
 #[derive(Copy, Clone, Eq, PartialEq)]
+enum Register {
+    CR1,
+    CR2,
+    OAR1,
+    OAR2,
+    TIMINGR,
+    TIMEOUTR,
+    ISR,
+    PECR,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, counters::Count)]
```

Comment:
now that this derives `Count`, we also need to change this like: https://github.com/oxidecomputer/hubris/blob/014b2a7fdd96506cbffe494bfb01adb26bc90598/drv/stm32xx-i2c/src/lib.rs#L207 to `counted_ringbuf!`

Reply:
We actually can't do that because there is no way (currently) to disable counted ringbufs entirely -- which we need to be able to do to fit in the Donglet.

Reply:
Ah, yup. I can go add that!

Reply:
Ah, it looks like, currently, the `ringbuf` crate's `counters` dependency _is_ optional, but that's not quite what we want, as I've learnt --- it's preferable to be able to opt _out_ of instrumentation on small targets rather than opt in on all the other targets; thus, "`ringbuf/disabled"`. I'll add a similar feature flag for disabling counts as well.

Reply:
Okay, #1658 will let you disable both the actual ringbuf and counters for ringbufs, so we can add a `counted_ringbuf!` here and still fit in the STM32G031's RAM.

Reply:
Merged #1658, so we can make this counted now!